cypress/integration/pages/frontPage/tests.js

```diff
@@ -92,26 +92,50 @@ export const testsThatFollowSmokeTestConfig = ({ service, pageType }) =>
           });
         });
 
-        if (serviceHasIndexAlsos(service)) {
-          it('should contain Index Alsos at a mobile view', () => {
-            cy.viewport('iphone-5');
-            cy.get('section li')
-              .eq(0)
-              .within(() => {
-                cy.get('div div div a')
+        it('should contain Index Alsos at a mobile and desktop view, if relatedItems block exists', () => {
+          cy.request(`${config[service].pageTypes.frontPage.path}.json`).then(
+            ({ body }) => {
+              const topstories = body.content.groups[0].items[0];
+              const relatedItemsExists = Object.prototype.hasOwnProperty.call(
+                topstories,
+                'relatedItems',
+              );
+
+              const runIndexAlsoTests = () => {
+                cy.get('[aria-labelledby="Top-stories"]')
                   .eq(0)
                   .within(() => {
-                    cy.get('span').then($el => {
-                      if ($el.length > 1) {
-                        cy.get('svg').should('be.visible');
-                      } else {
-                        expect($el).not.to.have.descendants('svg');
-                      }
-                    });
+                    cy.get('div')
+                      .eq(10)
```

Comment:
The 10th div? This seems really unstable IMO, is there anyway It can be made safer?

Reply:
My knowledge of Cypress is limited at best, but for now I don't see a different way of targeting the index alsos other than checking the HTML structure. So rather than doing `div` followed by a `div` within `ul` etc etc, I opted for this option. 

This is very flakey... and I don't like it, but afaik we can't really target by class because we're using styled components. We could use data attributes with `data-cy` which is preferred by Cypress, but that would require making changes to Psammead components.

Reply:
I'm happy to be proven wrong. It'd be great if there was an easy way of targeting the components we need to test, without having to really on just the HTML structure... It just makes tests flakey and unreadable

Reply:
So more on this, because I can't just let it go.

Created a PR in Psammead for adding the `data-cy` attribute - https://github.com/bbc/psammead/pull/2096, so now lines 108-109 can be changed to 
```
cy.get('[data-cy=index-alsos]')
 .eq(0)
```

Reply:
Out of curiosity do we use this anywhere else? `data-cy`

Reply:
Maybe this should be an id instead?

1) Doesn't lock us into cypress by adding cy values to the dom
2) The id's could be useful elsewhere

Reply:
Agreed. `id` looks much cleaner imo. Will update this PR and the Psammead one accordingly.

Reply:
I'd suggest keeping data attributes `data-cy` as a pattern for usage for cypress tests. This is because we should ideally only use ids for JS and anchor links. 
Keeping data attributes will be better in the long term - if we have multiple uses of index alsos on a page, having several `data-cy="index-alsos"` will be fine, but ids would clash - we'd have multiple of these on a page.

Reply:
turns out the solution is to just use class names 😄 thanks @sareh ⭐️